dev/puppetserver.conf

```diff
@@ -46,27 +46,27 @@ jruby-puppet: {
 
     # This setting determines where JRuby will install gems.  It is used for loading gems,
     # and also by the `puppetserver gem` command line tool.
-    gem-home: ./target/jruby-gems
+    gem-home: ${HOME}"/.puppetlabs/opt/server/data/puppetserver/jruby-gems"
```

Comment:
https://github.com/typesafehub/config/blob/master/HOCON.md#substitutions

Yeah, this is just how their syntax works.  It won't matter if there are special chars in their HOME var though; the interpolation happens in memory way after this file is parsed, and this is effectively just a string concatenation operation.